src/components/ui/drawers/TaskDeletionDrawer.tsx

```diff
@@ -52,39 +56,48 @@ export const TaskDeletionDrawer = ({
   title,
   open,
   setOpen,
+  taskId,
 }: ITaskDeletionDrawer) => {
-  const { handleSubmit } = useForm();
+  const path = usePathname();
 
-  const [isLoading, setIsLoading] = useState(false);
+  const {
+    handleSubmit,
+    formState: { isSubmitting },
+  } = useForm();
 
-  // TODO: Implement the delete click functionality
   const onSubmit = async () => {
-    setIsLoading(true);
     setOpen(true);
     await new Promise((resolve) => setTimeout(resolve, 1000));
 
-    if (title) {
-      console.log('title', title);
-      setIsLoading(false);
-      setOpen(false);
-      toast({
-        variant: 'default',
-        description: 'Updated successfully!',
+    try {
+      const res = await api.task[':taskId'].$delete({
+        param: {
+          taskId,
+        },
       });
-    } else {
-      setIsLoading(false);
-      setOpen(false);
+      const data = await res.json();
+      if ('error' in data) {
+        throw new Error(data.error);
+      }
       toast({
-        variant: 'destructive',
-        description: 'error!',
+        variant: 'default',
+        description: TOAST_TEXTS.success,
       });
+      revalidatePage(path);
+    } catch (error) {
+      if (error instanceof Error) {
+        toast({
+          variant: 'destructive',
+          description: error.message,
+        });
+      }
     }
   };
 
   return (
     <>
-      {isLoading ? <LoadingSpinner isLoading={true} /> : ''}
-      <Drawer open={open} onOpenChange={setOpen} modal={!isLoading}>
+      {isSubmitting ? <LoadingSpinner isLoading={true} /> : ''}
+      <Drawer open={open} onOpenChange={setOpen}>
```

Comment:
Could you re-add `modal={!isSubmitting}`?

Reply:
Ohhh thank you for letting me know